### DISCUSSION TITLE
Build times are up 10x after 3.1.5

### DISCUSSION THREAD

**Discussion by DanielGSoftware** [tailwindlabs/tailwindcss#9873] [Help] [answered]
Hi there!

We've recently updated our Tailwind to the latest version, 3.2.4. However, this caused our npm build times (dev & production) to take around 10x times more time to complete. For example, 10 seconds on 3.1.4 compared to 130 seconds on 3.1.5 and above. 
Also our tailwind.css file increased from 624KiB to 5.35 MiB.

To narrow the cause down, I started by upgrading to the latest Tailwind version step by step. On the first upgrade (3.1.5) I already found that this causes the slow build times. I've looked at each PR but as mostly being a back-end developer I did not manage to find the issue for us.

Could someone with more knowledge please help us on this? 
Thanks a lot in advance :)

**Comment by DanielGSoftware** [comment, marked as answer]
After some more debugging, I think it's related to https://github.com/tailwindlabs/tailwindcss/pull/8774. We have the following in our safelist:
```js
{
    pattern: /border-/,
},
```
<br>

Replaced that with 
```js
{
    pattern: /^border-.*-200$/,
},
```
brings the build time back to normal.

**Comment by chojnicki** [comment]
@DanielGSoftware you saved me a lot of time on debugging. What a coincidence that we had exact same problem in exact same time. Thank you.

**Reply by DanielGSoftware** [reply]
Coincidental indeed, you're welcome :)
